src/store/backendApi.ts

```diff
@@ -43,6 +43,28 @@ export const useLazyGetOscapCustomizationsQuery = process.env.IS_ON_PREMISE
   ? cockpitQueries.useLazyGetOscapCustomizationsQuery
   : serviceQueries.useLazyGetOscapCustomizationsQuery;
 
+/**
+ * Hook to fetch OSCAP customizations for a specific compliance policy.
+ * Returns policy-specific customizations (packages, services, kernel args) that
+ * may differ from the base profile due to policy tailoring.
+ * 
+ * @returns undefined for on-premise environments (falls back to profile-based customizations)
+ */
+export const useGetOscapCustomizationsForPolicyQuery = process.env.IS_ON_PREMISE
```

Comment:
We could probably rename this since we try to differentiate between `Oscap` and `Compliance`,
something like `useGetCompliancCustomizationsQuery`